### PR TITLE
 Revamp azure event hub client and vSLAlpha4 migration

### PR DESCRIPTION
**Pull request by RolandHewage**
- The Management client & publisher client should be merged to a single client to support Choreo UI
- We need to provide record types instead of XML responses from remote operations
- Remove default values from optional parameters & let it handle on the server side.
- We need to return a stream of records for XML lists.

Resolves https://github.com/wso2-enterprise/choreo/issues/3399
Resolves https://github.com/wso2-enterprise/choreo/issues/3595

Migrate the connector to SLAlpha4 and add display annotations

- Merge Management client & publisher client to a single client to support Choreo UI
- Provide record types instead of XML responses from remote operations
- Remove default values from optional parameters & let it handle on the server side.
- Return a stream of records for XML lists.
- Add test cases for the changes and automate them
- Update samples & documentation

## Release note
- Merge Management client & publisher client to a single client to support Choreo UI
- Provide record types instead of XML responses from remote operations
- Remove default values from optional parameters & let it handle on the server side.
- Return a stream of records for XML lists.

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

JDK 11
Ubuntu 20.04
Service Event Hub Connector Version 0.1.2
 
https://azure.microsoft.com/en-us/services/event-hubs/#overview
https://docs.microsoft.com/en-us/rest/api/eventhub/
https://docs.microsoft.com/en-us/azure/event-hubs/authenticate-shared-access-signature

### CODE REVIEW

File: data_mappings.bal [hunk 1]
```diff
@@ -45,14 +46,270 @@ isolated function getBatchEventJson(BatchEvent batchEvent) returns json {
...
+    json idElementEH = check xmldata:toJson(xmlContent/<id>);
+    json idEH = check idElementEH.id.id; 
+    consumerGroup.id = idEH.toString();
+    json titleElementEH = check xmldata:toJson(xmlContent/<title>);
+    json titleEH = check titleElementEH.title.title; 
+    consumerGroup.title = titleEH.toString();
+    json publishedElementEH = check xmldata:toJson(xmlContent/<published>);
+    json publishedEH = check publishedElementEH.published.published;
+    consumerGroup.published = publishedEH.toString();
+    json updatedElementEH = check xmldata:toJson(xmlContent/<updated>);
```
Comment (abeykoon) [inline data_mappings.bal:243]: We can access value of xml element tags by <path to leaf element>/* 
Shall we check on that approach?

Comment (RolandHewage) [reply data_mappings.bal]: Resolved in 6c77321 . Thank you.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Ballerina Build
-        uses: ballerina-platform/ballerina-action/@swan-lake-connector-release
+        uses: ballerina-platform/ballerina-action/@master
         with:
           args:
             build -c --skip-tests
```
Comment (SanduDS) [inline .github/workflows/release.yml:18]: Need to run tests when releasing? it's ok to skip for specific reason? but the ballerina issue is fixed now

Comment (RolandHewage) [reply .github/workflows/release.yml]: Resolved in 57afa94. Thank you.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**